docs/preview/logdecorations.md

```diff
@@ -0,0 +1,260 @@
+
+# Log decorations
+
+
+Task authors should be able to control how the log output is displayed to the end user.
+This outlines different decoration options that are available.
+
+## Special lines
+* Errors
+  * `##[error] I am an error` 
+* Warnings
+  * `##[warning] I am a warning` 
+* Commands
+  * `##[warning] I am a command/a tool` 
+* Sections
+  * `##[section] I am a section, which is usually whole task step. I am not typically used.` 
+
+## Collapse
+
+>This is a draft spec and is subject to change
+
+>Note that that if you log an error using ```##vso[task.logissue]error/warning message``` command (see [logging commands](https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md) here) we will surface those errors in build view and when clicked , we will automatically jump to that particular line. If it's already part of a group, we will auto-expand the group.
+
+Task authors can mark any part of the log as a collapsible region using these decorations:
+
+Starting the collapsible region - `##[startgroup<optionalGroupName>]`
+
+Ending the collapsible region - `##[endgroup<optionalGroupName>]`
+
+`<optionalGroupName>` is optional.
+
+Note: 
+    
+*  The first line of region will be taken as group title by default.
+*  If there's only one line in the region (including the group title), it will not be considered as a collapsible
+*  If there's `##[startgroup]` with out corresponding `##[endgroup]` we will add implicit `##[endgroup]`
+   *  This applies to only non-named groups, if group name is specified, no implcit  `##[endgroup]` is considered until the end of the content.
+
+Example 1 -
```

Comment:
for ad hoc scripting, i suggest optimizing for simplicity over robustness.

```
echo ##[group] testing with node 5
...
echo ##[group] testing with node 6
...
```

for finer control, allow `endgroup`

but only use the name `group` instead of `startgroup` since it aligns better with simple scripting and is easier on the eyes.

Reply:
yeah, I like just `##[group<:name>]` instead of `##[startgroup<:name>]`
and have `##[endgroup<:name>]` as optional

Reply:
we are doing `##[group]`